cmd/zoekt-sourcegraph-indexserver/fetcher.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/google/zoekt"
+	"io"
+	"os/exec"
+	"strings"
+)
+
+// a deltaFetcher returns a list of changed files along with their file contents
+//
+// NOTE: The idea of returning the file contents in the same function seems awkward
+// with local checkouts, but necessary in a remote one
+type deltaFetcher interface {
+	// Fetch returns the paths that have changed on the latest
+	// commit for each branch, along with their file contents.
+	Fetch(ctx context.Context, branches ...zoekt.RepositoryBranch) (branchDeltas, error)
```

Comment:
we probably need to represent in the return interface us failing due to the response being too large. IE fallback to doing a fullclone. This probably shouldn't be represented with an error since that makes it too easy to ignore.